src/native/eventpipe/CMakeLists.txt

```diff
@@ -12,7 +12,9 @@ check_symbol_exists(
     HAVE_ACCEPT4)
 
 # Use TCP for EventPipe on mobile platforms
-if (CLR_CMAKE_HOST_IOS OR CLR_CMAKE_HOST_TVOS OR CLR_CMAKE_HOST_ANDROID)
+#
+# TODO: https://github.com/dotnet/runtime/issues/111524
+if (CLR_CMAKE_HOST_IOS OR CLR_CMAKE_HOST_TVOS OR (CLR_CMAKE_HOST_ANDROID AND NOT CLR_CMAKE_RUNTIME_CORECLR))
```

Comment:
Is there a specific reason why this is disabled? I fixed the build for CoreCLR/iOS in https://github.com/dotnet/runtime/pull/109928. Perhaps it's the same underlying error?

Reply:
Could be. Worth looking into.

Reply:
I verified that the change in `src/native/eventpipe/ds-ipc-pal-socket.c` from #109928 fixes the Android build as well and this would no longer be needed.